code/game/objects/random/produce.dm

```diff
@@ -0,0 +1,44 @@
+/obj/random_produce
+	name = "random produce"
+	icon = 'icons/obj/seeds.dmi'
+	icon_state = ""
+	var/list/produce_list = list(
+		"chili" = 1,
+		"berry" = 0.25,
+		"blueberry" = 0.25,
+		"tomato" = 2,
+		"eggplant" = 0.5,
+		"apple" = 0.25,
+		"grapes" = 0.25,
+		"green grape" = 0.25,
+		"peanut" = 0.5,
+		"cabbage" = 2,
+		"banana" = 0.5,
+		"corn" = 2,
+		"potato" = 2,
+		"soybean" = 0.5,
+		"rice" = 2,
+		"carrots" = 1,
+		"white-beet" = 1,
+		"watermelon" = 0.25,
+		"pumpkin" = 0.25,
+		"lime" = 0.25,
+		"lemon" = 0.25,
+		"orange" = 0.25,
+		"cocoa" = 0.5,
+		"cherry" = 0.25,
+		"garlic" = 0.5,
+		"onion" = 0.5
+	)
+
+/obj/random_produce/Initialize(var/mapload = 1)
```

Comment:
Why are you specifying a default value for mapload?

Reply:
what do you think i know how to code